src/python/graphit.py

```diff
@@ -84,6 +84,101 @@ def compile_and_load(graphit_source_file, extern_cpp_files=[], linker_args=[], p
 
 	return module
 
+def compile_and_load_cache(graphit_source_file, extern_cpp_files=[], linker_args=[], parallelization_type=PARALLEL_NONE):
+	graphit_source_file = os.path.expanduser(graphit_source_file.strip())
+	# Obtain a unique filename for the module
+	#module_file = tempfile.NamedTemporaryFile()
+	#module_filename_base = module_file.name
+	#module_file.close()
+	module_filename_base = os.path.splitext(graphit_source_file)[0]
+	module_name = os.path.basename(module_filename_base)    
+	module_filename_base = "/tmp/" + module_name
+
+	# compile the file into a cpp file
+	module_filename_cpp = module_filename_base + ".cpp"
+	module_filename_object = module_filename_base + ".o"
+	module_filename_so = module_filename_base + ".so"
+
+	# check if it is loaded before and it is the latest, we return the cached version.
+	if os.path.exists(module_filename_so) and os.stat(graphit_source_file).st_mtime < os.stat(module_filename_so).st_mtime:
+		spec = importlib.util.spec_from_file_location(module_name, module_filename_so)
+		module = importlib.util.module_from_spec(spec)
+		spec.loader.exec_module(module)
+		return module
+	
+	#else we just follow the normal routine
+	try:
+		subprocess.check_call("python " + GRAPHIT_BUILD_DIRECTORY + "/bin/graphitc.py -f " + graphit_source_file + " -o " + module_filename_cpp + " -m " + module_name, stderr=subprocess.STDOUT, shell=True)
```

Comment:
Can you add some unit tests for the compile_and_load_cache method in pybind_test.py (https://github.com/GraphIt-DSL/graphit/blob/master/test/python/pybind_test.py)? And try to add some unit tests for the two files in the apps/python_binding directory in pybind_test.py?